### DISCUSSION TITLE
mcu unable to connect

### DISCUSSION THREAD

**Discussion by Monsinia** [Guilouz/Klipper-Flsun-Speeder-Pad#98] [Q&A]
I have looked at several other posts and I can't find anything that resembles my problem.  
I have followed every step on the guide and am at the "Update Motherboard Firmware)

When I go to Mainsail to look for my firmware version I only get:
"Klipper reports: ERROR 
mcu 'mcu' : Unable to connect
Once the underlying issue is corrected, use the
"FIRMWARE_RESTART" command to reset the firmware, reload the 
config, and restart the host software.
Error configuring printer"

I have followed every step successfully up to this point.   Did I potentially miss something? and if so how can I reset everything back to factory settings so that I can actually use the printer in the event I dont figure this out? 

Thanks

![image](https://user-images.githubusercontent.com/130875995/232259713-74818552-8218-4e2b-ab4a-187462458116.png)

**Comment by coinmycoke** [comment]
I same issue here... had a running MAILSAIL on a Raspberry that worked so I did not do the firmware update. 

![image](https://user-images.githubusercontent.com/73829275/232308034-564f02ab-e1f0-41a1-bca9-66dfda0d1e0f.png)

**Comment by Fropzs** [comment]
What is the mcu in your printer config?

**Reply by coinmycoke** [reply]
Got it solved somehow my conf lost the MCU definition. 
Now it works.

**Reply by Hsubsy71** [reply]
how did you fix it I have the same problem

**Reply by Fropzs** [reply]
> how did you fix it I have the same problem

Chek your printer config and see if the mcu is changed

**Reply by Monsinia** [reply]
Sorry but can you explain the solution to me like I’m five?

thanks

**Reply by Fropzs** [reply]
if you have the same problem.

go on mainsail - machine-config files-printer.cfg 

scroll down to you see your MCU Settings

See if your MCU is somthing like this: 
[mcu]
serial: /dev/serial/by-id/usb-1a86_USB_Serial-if00-port0
restart_metho: command

**Comment by Monsinia** [comment]
Okay, I finished my project after restoring everything to factory.  so i'll have to go through the entire process again up to that point.  I'll comment again when I get to it.  Thanks

________________________________

**Reply by c3cerberus** [reply]
So what he is talking about: 

Open up your Mainsail Web page.
Go to Machine
Select Printer.cfg

Does is look like this:
[mcu]
serial: /dev/serial/by-id/<your-mcu-id>

or does it look something like this:

[mcu]
serial: /dev/serial/by-id/usb-1a86_USB_Serial-if00-port0
restart_metho: command

???

If it is the first solution, your printer has dropped the config file.

**Reply by c3cerberus** [reply]
If your Printer dropped its confg, i.e.  (Serial number), do this:

Connect your printer to one of the Speeder Pad's USB ports.

Connect in SSH then enter the following command to retrieve the motherboard USB serial:

cd ~ && ls /dev/serial/by-id/*

o to your Mainsail Web interface then click on Machine tab.

Open printer.cfg file and find [mcu] section.

Edit the following existing line with serial you have just obtained like this:

serial: /dev/serial/by-id/usb-1a86_USB_Serial-if00-port0
Click on SAVE & RESTART at the top right to save the file.

Your printer should connect to your Speeder Pad.

THIS IS FOR A SINGLE INSTANCE USE CASE. IF YOU DID MULTIPLE INSTANCES THERE IS ANOTHER STEP. LET ME KNOW.

**Reply by Superexe1** [reply]
didnt work for me...been working on this for a few days..lost
![scrnli_5_6_2023_11-36-05 AM](https://user-images.githubusercontent.com/122970910/236633619-f41f832e-60b1-424a-88f2-9d267b5bd425.png)
![scrnli_5_6_2023_11-36-33 AM](https://user-images.githubusercontent.com/122970910/236633621-728bd3f5-9fd6-4825-b6a8-467502850249.png)

**Reply by Superexe1** [reply]
I THINK I GOT IT..update the firmware on the printer fixed it for me

**Comment by Monsinia** [comment]
Awesome!  I will refer back this when i go through the process again.



________________________________
